source/verus/src/main.rs

```diff
@@ -75,19 +51,263 @@ fn main() {
+    if !reports_dir.clone().join(".git").is_file() {
+        std::fs::create_dir_all(reports_dir.clone())
+            .expect("Creating verus/ directory in local data cache");
+
+        std::process::Command::new("git")
+            .current_dir(&reports_dir)
+            .arg("init")
+            .output()
+            .expect("Initializing git in local data cache");
+    }
+
+    let uuid_file = reports_dir.join("userid");
+    if !uuid_file.is_file() {
+        let uuid = uuid::Uuid::new_v4();
+        std::fs::write(uuid_file.clone(), uuid.to_string()).expect("Writing user id file");
+    }
+
+    let uuid = std::fs::read_to_string(uuid_file).expect("Reading user id file");
+
+    // Step 3: project dir check/create (project = SHA256((normalized/absolute)path to root.rs)
+
+    // clone is not implemented for Args, so I grabbed the arguments here again
+    let new_args = std::env::args().into_iter();
+
+    let rs_files = new_args.filter(|arg| arg.ends_with(".rs")).collect::<Vec<_>>();
+
+    let input_file: Option<String> = match rs_files.len() {
+        1 => Some(rs_files[0].clone()),
+        _ => {
+            eprintln!("{}: cannot find input file", yansi::Paint::red("error"));
+            None // passed down to rust_verify for error message
+        }
+    };
+
+    let input_file_path = match input_file.clone() {
+        Some(input_file_name) => std::fs::canonicalize(input_file_name).ok(),
+        None => None,
+    };
+
+    let project_name = match input_file_path.clone() {
+        Some(path) => {
+            Some(format!("{:x}", sha2::Sha256::digest(path.to_str().unwrap().as_bytes())))
+        }
+        None => None,
+    };
+
+    let proj_dir = match project_name {
+        Some(name) => {
+            let project_dir = reports_dir.join(uuid).join(name);
+            if !project_dir.is_dir() {
+                std::fs::create_dir_all(project_dir.clone())
+                    .expect("failed to create project directory");
+            }
+            Some(project_dir)
+        }
+        None => None,
+    };
+
+    match (input_file_path, proj_dir) {
+        (Some(crate_root_path), Some(proj_dir)) => {
+            Some(Reports { proj_path: proj_dir, crate_root: crate_root_path })
         }
+        _ => None,
     }
 }
+
+fn get_dependencies(dep_file_path: &std::path::Path) -> Result<Vec<PathBuf>, String> {
+    // update to better error message
+    let file = std::fs::File::open(dep_file_path)
+        .map_err(|x| format!("{}, dependency file name: {:?}", x, dep_file_path))?;
+    let mut reader = std::io::BufReader::new(file);
+    let mut dependencies = String::new();
+    reader.read_line(&mut dependencies).map_err(|x| {
+        format!("Could not read the first line of the dependency file with message: {}", x)
+    })?;
+    let dep: Vec<String> = dependencies.split_whitespace().skip(1).map(|x| x.to_string()).collect();
+    println!("\n{} {:?}", yansi::Paint::blue("dependencies:"), dep);
```

Comment:
Looks like this is debug output for yourself; I don't think it should be included in the final code
```suggestion
```